lib/src/main/java/io/ably/lib/realtime/Channel.java

```diff
@@ -491,8 +525,10 @@ private void failQueuedMessages(ErrorInfo reason) {
 	 * @throws AblyException
 	 */
 	public PaginatedResult<Message> history(Param[] params) throws AblyException {
+		params = replacePlaceholderParams(this, params);
+
 		BodyHandler<Message> bodyHandler = MessageSerializer.getMessageResponseHandler(options);
-		return new PaginatedQuery<Message>(ably.http, basePath + "/history", HttpUtils.defaultAcceptHeaders(ably.options.useBinaryProtocol), params, bodyHandler).get();
+		return new PaginatedQuery<>(ably.http, basePath + "/history", HttpUtils.defaultAcceptHeaders(ably.options.useBinaryProtocol), params, bodyHandler).get();
```

Comment:
It is automatically using the generic passed down with `BodyHandler`. Therefore, we don't need to explicitly specify the generic again.